nibabel/cifti2/cifti2_axes.py

```diff
@@ -0,0 +1,1441 @@
+
+        Parameters
+        ----------
+        name : np.ndarray
+            (N, ) string array with the parcel names
+        voxels :  np.ndarray
+            (N, ) object array each containing a sequence of voxels.
+            For each parcel the voxels are represented by a (M, 3) index array
+        vertices :  np.ndarray
+            (N, ) object array each containing a sequence of vertices.
+            For each parcel the vertices are represented by a mapping from brain structure name to
+            (M, ) index array
+        affine : np.ndarray
+            (4, 4) array mapping voxel indices to mm space (not needed for CIfTI2 files only
+            covering the surface)
+        volume_shape : tuple of three integers
+            shape of the volume in which the voxels were defined (not needed for CIfTI2 files only
+            covering the surface)
+        nvertices : dict[String -> int]
+            maps names of surface elements to integers (not needed for volumetric CIfTI2 files)
+        """
+        self.name = np.asanyarray(name, dtype='U')
+        as_array = np.asanyarray(voxels)
+        if as_array.ndim == 1:
+            voxels = as_array.astype('object')
+        else:
+            voxels = np.empty(len(voxels), dtype='object')
+            for idx in range(len(voxels)):
+                voxels[idx] = as_array[idx]
+        self.voxels = np.asanyarray(voxels, dtype='object')
+        self.vertices = np.asanyarray(vertices, dtype='object')
+        self.affine = affine
+        self.volume_shape = volume_shape
+        if nvertices is None:
+            self.nvertices = {}
+        else:
+            self.nvertices = {BrainModel.to_cifti_brain_structure_name(name): number
+                              for name, number in nvertices.items()}
+
+        for check_name in ('name', 'voxels', 'vertices'):
+            if getattr(self, check_name).shape != (self.size, ):
+                raise ValueError("Input {} has incorrect shape ({}) for Parcel axis".format(
+                        check_name, getattr(self, check_name).shape))
+
+    @classmethod
+    def from_brain_models(cls, named_brain_models):
+        """
+        Creates a Parcel axis from a list of BrainModel axes with names
+
+        Parameters
+        ----------
+        named_brain_models : iterable of 2-element tuples of string and BrainModel
+            list of (parcel name, brain model representation) pairs defining each parcel
+
+        Returns
+        -------
+        Parcels
+        """
+        nparcels = len(named_brain_models)
+        affine = None
+        volume_shape = None
+        all_names = []
+        all_voxels = np.zeros(nparcels, dtype='object')
+        all_vertices = np.zeros(nparcels, dtype='object')
+        nvertices = {}
+        for idx_parcel, (parcel_name, bm) in enumerate(named_brain_models):
+            all_names.append(parcel_name)
+
+            voxels = bm.voxel[~bm.surface_mask]
```

Comment:
Might be more intuitive to switch from `~x.surface_mask` to `x.volume_mask`, now that it exists:

```suggestion
            voxels = bm.voxel[bm.volume_mask]
```